cdn/sign_url.rb

```diff
@@ -0,0 +1,63 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+def signed_url url:, key_name:, key_file:, expiration:
+  # [START signed_url]
+  # url        = "URL to content served by Google Cloud CDN"
+  # key_name   = "Signing key name"
+  # key_file   = "Signing key file name with a urlsafe base64 encoded key"
+  # expiration = "Expiration time for signed URL formatted as a Unix Timestamp"
+
+  require "base64"
+  require "cgi"
+  require "openssl"
+  require "uri"
+
+  # Read in url safe base64 encoded file
+  key         = File.read key_file
+  decoded_key = Base64.urlsafe_decode64 key_file
+
```

Comment:
`key_file` -> `key` ?

Reply:
`key_file` is the file path and `key` is the loaded value. Do you want me to hide read file in the sample?

Reply:
I'm confused, maybe I'm not reading this syntax correctly since I don't know Ruby, to me this is:

```
string key_file; // path of the file
key = file.read(key_file)
decoded_key = base64.urlsafe_decode64(key_file)
```

in the pseudo-code above (which I think is equivalent to this snippet?) `key` variable is unused and you're decoding the path string?

Reply:
Ah yea, that's wrong, I looked at the wrong line. Apologies.